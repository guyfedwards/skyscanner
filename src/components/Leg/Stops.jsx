import React from 'react';
import propTypes from 'prop-types';
import styled from 'styled-components';

const Root = styled.span`
  color: ${({ direct }) => (direct ? '#00a698' : '#d1435b')};
`;

const Stops = ({ stops }) => {
  const direct = stops === 0;
  const text = direct ? 'Direct' : `${stops} stop ${stops > 1 ? 's' : ''}`;

  return <Root direct={direct}>{text}</Root>;
};

Stops.propTypes = {
  stops: propTypes.number.isRequired,
};

export default Stops;
