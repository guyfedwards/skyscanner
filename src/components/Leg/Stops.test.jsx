import React from 'react';
import ReactDOM from 'react-dom';
import ShallowRenderer from 'react-test-renderer/shallow';

import Stops from './Stops';

describe('Stops', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<Stops stops={0} />, div);
  });

  it('should render correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Stops stops={1} />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
