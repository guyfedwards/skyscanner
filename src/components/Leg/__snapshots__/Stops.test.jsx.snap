// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stops should render correctly 1`] = `
<ForwardRef(styled.span)
  direct={false}
>
  1 stop 
</ForwardRef(styled.span)>
`;
